### PR TITLE
solution jv-fruit-shop-test

### PR DESCRIPTION
**Pull request by mykyta-udav**

### CODE REVIEW

File: src/test/java/core/basesyntax/service/impl/CsvFileReaderServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @BeforeAll
+    static void setUp() {
+        fileReaderService = new CsvFileReaderServiceImpl();
+    }
+
+    @Test
+    void readLines_filePathIsEmpty_notOk() {
+        String filePath = "";
+        assertThrows(RuntimeException.class, () -> fileReaderService.readLines(filePath));
```
Comment (MykytaKat) [inline src/test/java/core/basesyntax/service/impl/CsvFileReaderServiceImplTest.java:22]: please, add messages for more information. Fix it in all other tests

File: src/test/java/core/basesyntax/service/impl/CsvFileReaderServiceImplTest.java
Comment (MykytaKat) [inline src/test/java/core/basesyntax/service/impl/CsvFileReaderServiceImplTest.java:1]: you can add one more test, if input file is empty

File: src/test/java/core/basesyntax/service/impl/CsvFileWriterServiceImplTest.java
Comment (MykytaKat) [inline src/test/java/core/basesyntax/service/impl/CsvFileWriterServiceImplTest.java:1]: add more tests, for example, if file path is incorrectly listed

File: src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Test
+    void process_withNullInitializationList_notOk() {
+        List<FruitTransaction> fruitTransactions = null;
+        assertThrows(NullPointerException.class, ()
+                -> fruitService.processTransaction(fruitTransactions));
+    }
+
+    @Test
+    void process_withEmptyInitializationListShouldNotChangeTheStorage_ok() {
```
Comment (MykytaKat) [inline src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java:38]:
```suggestion
    void process_withEmptyInitializationList_ok() {
```

**Review by MykytaKat — CHANGES_REQUESTED** [review]
good job. let's improve your solution

**Review by MykytaKat — APPROVED** [review]
great! keep pushing

File: src/test/java/core/basesyntax/service/strategy/OperationStrategyTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    static void setUp() {
+        productDao = new ProductDaoImpl();
+        operationStrategy = new OperationStrategy(productDao);
+    }
+
+    @Test
+    void getOperation_getBalanceOperation_ok() {
+        OperationHandler expected = new BalanceOperationHandler(productDao);
+        OperationHandler actual = operationStrategy.getOperation(
+                new FruitTransaction(Operation.BALANCE, "banana", 10));
```
Comment (overpathz) [inline src/test/java/core/basesyntax/service/strategy/OperationStrategyTest.java:30]: In several test cases you use identical data - Operation.BALANCE, "banana", 10 -- Move it to constants

File: src/test/java/core/basesyntax/service/impl/ParserServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.FruitTransaction.Operation;
+import core.basesyntax.service.ParserService;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ParserServiceImplTest {
```
Comment (overpathz) [inline src/test/java/core/basesyntax/service/impl/ParserServiceImplTest.java:14]: Consider add tests for invalid arguments
p,strawberry-10

File: src/test/java/core/basesyntax/service/strategy/OperationStrategyTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import core.basesyntax.model.FruitTransaction.Operation;
+import core.basesyntax.service.impl.ProductDaoImpl;
+import core.basesyntax.service.strategy.impl.BalanceOperationHandler;
+import core.basesyntax.service.strategy.impl.PursheOperationHandler;
+import core.basesyntax.service.strategy.impl.ReturnOperationHandler;
+import core.basesyntax.service.strategy.impl.SupplyOperationHandler;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class OperationStrategyTest {
```
Comment (overpathz) [inline src/test/java/core/basesyntax/service/strategy/OperationStrategyTest.java:16]: What if we pass not existing strategy? (null, etc)
smth like getNotExistOperation_NotOk

**Review by overpathz — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        FruitStorage.Storage_Map.put("banana", 0);
+        int expectedQuantityOfApple = 100;
+        int expectedQuantityOfBanana = 100;
+        List<FruitTransaction> fruitTransactions = List.of(
+                new FruitTransaction(Operation.RETURN, "banana", 100),
+                new FruitTransaction(Operation.RETURN, "apple", 100));
+        fruitService.processTransaction(fruitTransactions);
+        int actualQuantityOfApple = FruitStorage.Storage_Map.get("apple");
+        int actualQuantityOfBanana = FruitStorage.Storage_Map.get("banana");
+        assertEquals(expectedQuantityOfApple, actualQuantityOfApple);
```
Comment (maxline) [inline src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java:63]: you can use `assertIterableEquals()` for asserting list values

File: src/test/java/core/basesyntax/service/strategy/OperationStrategyTest.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+    @BeforeAll
+    static void setUp() {
+        productDao = new ProductDaoImpl();
+        operationStrategy = new OperationStrategy(productDao);
+    }
+
+    @Test
+    void getOperation_getBalanceOperation_ok() {
+        OperationHandler expected = new BalanceOperationHandler(productDao);
```
Comment (maxline) [inline src/test/java/core/basesyntax/service/strategy/OperationStrategyTest.java:29]:
```suggestion
        Class<BalanceOperationHandler> expected = BalanceOperationHandler.class;
```
please fix in similar way in the other test methods.

File: src/test/java/core/basesyntax/service/strategy/OperationStrategyTest.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        OperationHandler actual = operationStrategy.getOperation(
+                new FruitTransaction(Operation.BALANCE, "banana", 10));
+        assertEquals(expected.getClass(), actual.getClass());
```
Comment (maxline) [inline src/test/java/core/basesyntax/service/strategy/OperationStrategyTest.java:32]:
```suggestion
        assertEquals(expected, actual.getClass());
```

**Review by maxline — CHANGES_REQUESTED** [review]
Good job! Let's improve the solution

File: src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java (hunk 6 above)
Comment (mykyta-udav) [reply src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java:63]: I think I don't need to check the value because I'm checking the operation

**Review by sarakhmen — APPROVED** [review]
Good job ;)
